ci/kind/kind-setup.sh

```diff
@@ -244,25 +245,66 @@ function configure_extra_networks {
   done
 }
 
+# add_kind_ipam_routes add routes for non-ipam test-pods.
+function add_kind_ipam_routes {
+  kubectl get nodes -o jsonpath='{range .items[*]}{.spec.podCIDR}{" "}{.status.addresses[?(@.type=="InternalIP")].address}{"\n"}{end}' |\
+  while read pod_cidr node_ip; do
+    docker_run_with_host_net ip route add "$pod_cidr" via "$node_ip"
+  done
+}
+
+# delete_kind_ipam_routes delete the routes of non-ipam test-pods from the Kind testbed.
+function delete_kind_ipam_routes { 
+  cluster=$(kind get clusters)
+  if [ ! -z "$cluster" ]; then
+    cidrs=$(kubectl get nodes -o jsonpath='{.items[*].spec.podCIDR}')
+    echo -e "$cidrs" | tr ' ' '\n'| while read cidr; do
+      del_routes=$(ip route | awk -v cidr="$cidr" '$0 ~ cidr {print$1}')
+      if [[ -n $del_routes ]]; then
+        docker_run_with_host_net ip route del $del_routes || true
+      fi 
+    done
+  fi
+}
+
 function configure_vlan_subnets {
   if [[ -z $VLAN_SUBNETS || -z $VLAN_ID ]]; then
     return
   fi
   echo "Configuring VLAN subnets"
 
-  bridge_id=$(docker network inspect kind -f {{.ID}})
-  bridge_interface="br-${bridge_id:0:12}"
-  vlan_interface="br-${bridge_id:0:7}.$VLAN_ID"
-
-  docker_run_with_host_net ip link add link $bridge_interface name $vlan_interface type vlan id $VLAN_ID
-  docker_run_with_host_net ip link set $vlan_interface up
   IFS=',' read -r -a vlan_subnets <<< "$VLAN_SUBNETS"
-  for s in "${vlan_subnets[@]}" ; do
-    echo "Configuring extra IP $s to vlan interface $vlan_interface"
-    docker_run_with_host_net ip addr add dev $vlan_interface $s
+  IFS=',' read -r -a vlan_ids <<< "$VLAN_ID"
+  i=0
+  single_vlan=false
+  
+  for s in "${vlan_subnets[@]}"; do
+    if [[ $single_vlan == true ]]; then
+      break
+    fi
+
+    vlan_id=${vlan_ids[$i]}
+    i=$(($i+1))
+    echo "Configuring VLAN-$vlan_id subnets"
+    bridge_id=$(docker network inspect kind -f {{.ID}})
+    bridge_interface="br-${bridge_id:0:12}"
+    vlan_interface="br-${bridge_id:0:7}.$vlan_id"
+
+    docker_run_with_host_net ip link add link $bridge_interface name $vlan_interface type vlan id $vlan_id
+    docker_run_with_host_net ip link set $vlan_interface up
+    if [[ ${#vlan_ids[@]} == 1 ]]; then
+      for s in "${vlan_subnets[@]}" ; do
+        echo "Configuring extra IP $s to vlan interface $vlan_interface"
+        docker_run_with_host_net ip addr add dev $vlan_interface $s
+      done
+        single_vlan=true
+    else
```

Comment:
This would make the meaning of VLAN_SUBNETS and VLAN_ID weird and inconsistent when called with different parameters.
If no good way, maybe merge vlan and subnet in a parameter that can be specified multiple times, like `--vlan-subnets <vlan-id>=<subnet-1>,<subnet-2>`.
Then:
* `--vlan-subnets 10=1.1.1.0/24,1234::/64` can be used by dual stack case
* `--vlan-subnets 10=1.1.1.0/24 --vlan-subnets 11=1.1.2.0/24` can be used by flexible IPAM.

Reply:
no need to use --vlan-id flag to pass vlan id?